File: packages/core/stages/05_translate/args.ts
import { z } from "zod/v4";
import { langList } from "../../const/lang";

export const TranslateCliInputSchema = z
  .looseObject({
    apiBase: z.string().optional(),
    model: z.string().optional(),
    targetLang: z
      .enum(langList)
      .optional()
      .describe("如果不填则 按照这个逻辑: 源语言: zh -> en, 否则 any -> zh"), //
    enabled: z.boolean().default(true).describe("设为 false 跳过翻译，直接使用原始识别文本"),
  })
  .prefault({});

File: packages/core/src/stages/translate/args.rs
use serde::{Deserialize, Serialize};

use config_rs::env::{openai_base_url, openai_model};

/// translate 阶段参数 (镜像 TS `packages/core/stages/05_translate/args.ts` TranslateArgsSchema)
///
/// 枚举/字符串默认值 TS 在写入 ctx.json 前已落定 (zod `.prefault({})` / `.default(...)`),
/// 这里只需处理「对象存在但字段缺」: 字段级 `#[serde(default…)]` 兜底即可。
///
/// 目标语言统一在 `input.workflow.targetLang` 配置 (任务级概念: 一个任务只有一个翻译
/// 阶段, 不需要 stage 级覆盖)。旧的 `stages.translate.targetLang` 已移除,
/// deny_unknown_fields 让残留配置明确报错 (不能静默失效 -> 翻错语言)。
#[derive(Debug, Clone, Default, Serialize, Deserialize, specta::Type)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct TranslateArgs {
    /// OpenAI 兼容端点
    #[serde(default = "openai_base_url")]
    pub api_base: String,
    /// 翻译模型
    #[serde(default = "openai_model")]
    pub model: String,
    /// 设为 false 跳过翻译, 直接使用原始识别文本
    #[serde(default = "default_true")]
    pub enabled: bool,
}

fn default_true() -> bool {
    true
}
